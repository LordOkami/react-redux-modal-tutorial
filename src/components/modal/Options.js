import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { selectOption } from "../../routines";

const Options = ({ modal: { options, selectedOption }, selectOption }) => (
  <React.Fragment>
    {options &&
      options.map((option, index) => (
        <div
          key={index}
          className={`option ${selectedOption == index && "selected"}`}
          onClick={() => selectOption(index)}
        >
          {option}
        </div>
      ))}
  </React.Fragment>
);

Options.propTypes = {
  modal: PropTypes.object,
  selectOption: PropTypes.func
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ selectOption }, dispatch)
});

const mapStateToProps = ({ modal }) => ({
  modal
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Options);
