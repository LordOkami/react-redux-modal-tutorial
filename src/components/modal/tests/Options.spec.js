import React from "react";
import { shallow } from "enzyme";
import { Options } from "../Options";

describe("<Options />", () => {
  const modal = { open: true, options: ["uno", "dos"], selectedOption: 0 };
  const selectOption = jest.fn;
  const props = { modal, selectOption };
  it("matches the latest snapshot", () => {
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
