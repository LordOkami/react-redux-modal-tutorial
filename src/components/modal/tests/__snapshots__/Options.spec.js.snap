// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> matches the latest snapshot 1`] = `
<Fragment>
  <div
    className="option selected"
    key="0"
    onClick={[Function]}
  >
    0
     - 
    uno
  </div>
  <div
    className="option false"
    key="1"
    onClick={[Function]}
  >
    1
     - 
    dos
  </div>
</Fragment>
`;
